### PR TITLE
 ignore and warn adding duplicate component, #993

### PR DESCRIPTION
**Pull request by Zhack47**

### CODE REVIEW

**Comment by AlmasB** [comment]
Well spotted, thanks.

@adambocco please can you review when you get a chance

File: fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ class EntityTest {
 
     @Test
     fun `Add component fails if same type already exists`() {
-        val exception = assertThrows(IllegalArgumentException::class.java) {
+        val exception = assertDoesNotThrow {
```
Comment (adambocco) [inline fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt:52]: The test name doesn't match the test performed in this change. Should we just get rid of this test and make one like `Adding duplicate component is ignored` and check if the original component remains on the entity.

File: fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt [hunk 2]
```diff
@@ -170,14 +170,15 @@ class EntityTest {
         assertThat(entity.components, hasItem(control))
     }
 
+    // Test seems duplicate, see l.51 `Add component fails if same type already exists`
```
Comment (adambocco) [inline fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt]: Same here. Maybe change this to create two `TestControl` objects and check if the first one is the added component like:
```
control1 = TestControl()
control2 = TestControl()
entity.addComponent(control1)
entity.addComponent(control2)
assertThat(entity.getComponent(TestControl::class.java),  `is`(control1))
```

**Review by adambocco — CHANGES_REQUESTED** [review]
The change looks good as long as it seems like the best idea to warn and ignore instead of throwing an error for duplicate components. I could see a case for either side but would rather the ignore as well.

**Review by fizietechD — APPROVED** [review]

**Review by fizietechD — APPROVED** [review]

File: fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt (hunk 1 above)
Comment (AlmasB) [reply fxgl-entity/src/test/kotlin/com/almasb/fxgl/entity/EntityTest.kt:52]: Good point, updated.

**Review by AlmasB — APPROVED** [review]
Thanks for the reviews @adambocco @Fizie92 

Thanks for the PR @Zhack47

**Pull request merged**